File: src/test/resources/projects/testJavaNoParamProject/reference/src/test/java/com/company/bonitasoft/api/IndexTest.java
package com.company.bonitasoft.api;

import org.bonitasoft.web.extension.rest.RestAPIContext;
import com.company.bonitasoft.api.dto.Result;
import org.bonitasoft.web.extension.ResourceProvider;
import org.bonitasoft.web.extension.rest.RestApiResponse;
import org.bonitasoft.web.extension.rest.RestApiResponseBuilder;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;
import org.mockito.junit.jupiter.MockitoSettings;
import org.mockito.quality.Strictness;

import jakarta.servlet.http.HttpServletRequest;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.time.LocalDate;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
@MockitoSettings(strictness = Strictness.LENIENT)
class IndexTest {

    // Declare mocks here
    // Mocks are used to simulate external dependencies behavior
    @Mock
    private HttpServletRequest httpRequest;
    @Mock
    private ResourceProvider resourceProvider;
    @Mock
    private RestAPIContext context;

    // The controller to test
    private Index index;

    /**
     * You can configure mocks before each tests in the setup method
     */
    @BeforeEach
    void setUp() throws FileNotFoundException {
        // Create a new instance under test
        index = new Index();

        // Simulate access to configuration.properties resource
        when(context.getResourceProvider()).thenReturn(resourceProvider);
        when(resourceProvider.getResourceAsStream("configuration.properties"))
                .thenReturn(IndexTest.class.getResourceAsStream("/testConfiguration.properties"));
    }


    @Test
    void should_get_result_when_params_ok() {

        // Given

        // When
        Result result = index.execute(context);

        // Then
        assertThat(result.getMyParameterKey()).isEqualTo("testValue");
        assertThat(result.getCurrentDate()).isEqualTo(LocalDate.now());
    }

    @Test
    void should_return_a_json_representation_as_result() throws IOException {
        // Given "a RestAPIController"

        // Simulate a request with a value for each parameter

        // When "Invoking the REST API"
        RestApiResponse apiResponse = index.doHandle(httpRequest, new RestApiResponseBuilder(), context);

        // Then "A JSON representation is returned in response body"
        Result jsonResponse = index.getMapper().readValue((String) apiResponse.getResponse(), Result.class);

        // Validate returned response
        assertThat(apiResponse.getHttpStatus()).isEqualTo(200);
        assertThat(jsonResponse).isNotNull();
    }

}

File: src/test/resources/projects/testKotlinNoParamProject/reference/src/test/kotlin/com/company/bonitasoft/IndexTest.kt
package com.company.bonitasoft

import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import jakarta.servlet.http.HttpServletRequest
import org.bonitasoft.web.extension.ResourceProvider
import org.bonitasoft.web.extension.rest.RestApiResponseBuilder
import org.bonitasoft.web.extension.rest.RestAPIContext
import java.time.LocalDate

class IndexTest {

    // Declare mocks here
    // Mocks are used to simulate external dependencies behavior
    val httpRequest = mockk<HttpServletRequest>()
    val resourceProvider = mockk<ResourceProvider>()
    val context = mockk<RestAPIContext>()

    lateinit var index: Index

    @BeforeEach
    fun setUp() {
        // Create a new instance per test
        index = Index();

        // Simulate access to configuration.properties resource
        every { context.getResourceProvider() } returns resourceProvider;
        every { resourceProvider.getResourceAsStream("configuration.properties") } returns
                IndexTest::class.java.getResourceAsStream("/testConfiguration.properties");
    }

    @Test
    fun `should return a json representation as result`() {
        // Given


        // When
        val apiResponse = index.doHandle(httpRequest, RestApiResponseBuilder(), context)

        // Then
        val jsonResponse = index.mapper.readValue(apiResponse.response as String, Map::class.java)
        // Validate returned response
        assertThat(apiResponse.httpStatus).isEqualTo(200)
        assertThat(jsonResponse["myParameterKey"]).isEqualTo("testValue")
        assertThat(jsonResponse["currentDate"]).isEqualTo(LocalDate.now().toString())
    }

}